src/cockpit/389-console/src/lib/replication/replChangelog.jsx

```diff
@@ -25,21 +24,43 @@ export class Changelog extends React.Component {
             loading: false,
             errObj: {},
             showConfirmDelete: false,
-            saveOK: false,
+            saveBtnDisabled: true,
             // Changelog settings
-            clMaxEntries: this.props.clMaxEntries,
-            clMaxAge: this.props.clMaxAge.slice(0, -1),
-            clMaxAgeUnit: this.props.clMaxAge.slice(-1).toLowerCase(),
-            clTrimInt: this.props.clTrimInt,
+            clMaxEntries: Number(this.props.clMaxEntries) == 0 ? -1 : Number(this.props.clMaxEntries),
+            clMaxAge: Number(this.props.clMaxAge.slice(0, -1)) == 0 ? -1 : Number(this.props.clMaxAge.slice(0, -1)),
+            clMaxAgeUnit: this.props.clMaxAge != "" ? this.props.clMaxAge.slice(-1).toLowerCase() : "s",
+            clTrimInt: Number(this.props.clTrimInt) == 0 ? -1 : Number(this.props.clTrimInt),
             clEncrypt: this.props.clEncrypt,
             // Preserve original settings
-            _clMaxEntries: this.props.clMaxEntries,
-            _clMaxAge: this.props.clMaxAge.slice(0, -1),
-            _clMaxAgeUnit: this.props.clMaxAge.slice(-1).toLowerCase(),
-            _clTrimInt: this.props.clTrimInt,
+            _clMaxEntries: Number(this.props.clMaxEntries) == 0 ? -1 : Number(this.props.clMaxEntries),
+            _clMaxAge: Number(this.props.clMaxAge.slice(0, -1)) == 0 ? -1 : Number(this.props.clMaxAge.slice(0, -1)),
+            _clMaxAgeUnit: this.props.clMaxAge != "" ? this.props.clMaxAge.slice(-1).toLowerCase() : "s",
+            _clTrimInt: Number(this.props.clTrimInt) == 0 ? -1 : Number(this.props.clTrimInt),
```

Comment:
changelog fields are empty if encryption is changed and saved.

1. Set 'Changelog Maximum Entries' to 1000;
2. Click on Encryption checkbox;
3. Save Settings

Everything disappears and then appears but the Age and Interval fields are empty (they were '-1')

Reply:
![image](https://user-images.githubusercontent.com/5634892/128042183-612f789b-ac31-4d25-b327-40aee9cf25e2.png)

Reply:
Fixed, on a side note, do you like these number inputs?   They are new to PF4, otherwise it would be as text input of type "number" - like what we use everywhere ("Server Tuning" page, etc).